libs/liblobby/lobby/interface.lua

```diff
@@ -642,11 +650,41 @@ Interface.commandPattern["UPDATEBATTLEINFO"] = "(%d+)%s+(%S+)%s+(%S+)%s+(%S+)%s+
 function Interface:_OnClientBattleStatus(userName, battleStatus, teamColor)
 	local status = ParseBattleStatus(battleStatus)
 	status.teamColor = ParseTeamColor(teamColor)
+
 	self:_OnUpdateUserBattleStatus(userName, status)
+	if userName == self.myUserName then
+		self:_EnsureMyTeamNumberIsUnique()
+	end
 end
 Interface.commands["CLIENTBATTLESTATUS"] = Interface._OnClientBattleStatus
 Interface.commandPattern["CLIENTBATTLESTATUS"] = "(%S+)%s+(%S+)%s+(%S+)"
 
+function Interface:_EnsureMyTeamNumberIsUnique()
```

Comment:
This doesn't seem protocol specific, why isn't it in lobby?

Reply:
Protocol specific. ZKLS doesn't heave teamID https://github.com/ZeroK-RTS/Zero-K-Infrastructure/blob/a7dbf0afbe154ea7e87fa17aa77e9c91bf1b8626/Shared/LobbyClient/Protocol/Messages.cs#L546